v0/destinations/drip/transform.js

```diff
@@ -0,0 +1,238 @@
+const { EventType } = require("../../../constants");
+const {
+  getDestinationExternalID,
+  getFieldValueFromMessage,
+  constructPayload,
+  extractCustomFields,
+  isEmptyObject,
+  CustomError,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getErrorRespEvents,
+  getSuccessRespEvents
+} = require("../../util");
+const logger = require("../../../logger");
+const {
+  ENDPOINT,
+  identifyMapping,
+  trackMapping,
+  campaignMapping,
+  IDENTIFY_EXCLUSION_FIELDS,
+  TRACKING_EXLCUSION_FIELDS
+} = require("./config");
+const {
+  idValidity,
+  isValidEmail,
+  isValidTimestamp,
+  callValidity,
+  createUpdateUser
+} = require("./util");
+
+const identifyResponseBuilder = async (message, { Config }) => {
+  const { accountId } = Config.accountId;
+
+  const id = getDestinationExternalID(message, "dripId");
+  if (id) {
+    await idValidity(Config, id);
+  }
+
+  let email = getFieldValueFromMessage(message, "email");
+  if (!isValidEmail(email)) {
+    email = null;
+    logger.error("Email format is incorrect");
+  }
+  const userId = getFieldValueFromMessage(message, "userId");
+
+  let payload = constructPayload(message, identifyMapping);
+  payload.id = id;
+  payload.email = email;
+  payload.visitor_uuid = userId;
+
+  // id — >(yes) -> check if the user already exists (GET call): exists? continue: error;
+  // |
+  // (no)->email —>(yes) —> no need to check anything --> continue;
+  // 	|
+  // 	(no)—> visitor_uuid—>(yes) -> check if the visitor_uuid is present in the list: exists? continue: erorr;
+  // 				  |
+  // 				  (no)—> call can’t be made
+
+  const check = callValidity(id, email, userId, Config);
+  if (!check) {
+    throw new CustomError(
+      "dripId, email or userId is required for the call",
+      400
+    );
+  }
+  if (!payload.first_name && !payload.last_name) {
+    const name = getFieldValueFromMessage(message, "name");
+    if (name && typeof name === "string") {
+      const [fname, lname] = name.split(" ");
+      payload.first_name = fname;
+      payload.last_name = lname;
+    }
+  }
+
+  let customFields = {};
+  customFields = extractCustomFields(
```

Comment:
custom fields in the destination payload can be passed as an object for the key `custom_fields`. we should support that.